oop-special_class_method-medium-q11 (#699)
canvas-gamification/canvas-gamification-junit-tests: src/oop/special_class_method/medium/q11/QuestionText.md

# oop-special_class_method-medium-q11

## Question Text

Many appliances and electronic devices have an internal mechanism to moderate temperature. Make a Java class called
**Oven** that has attributes **currTemp**, **size**, and **colour** that specify different properties of the **Oven**.

### Oven Class:

Your `Oven` class already has the following items:

- Attributes:
    - A `double` called **currTemp**
    - A `double` called **size**
    - A `String` called **colour**
    - These are not visible from other classes

- Constructor:
    - An `Oven` constructor that is visible to other classes and accepts two `double` and a `String` and uses it to
      initialize the **currTemp**, **size**, and **colour** attributes.

> - Methods:

Reviewer: fix spacing

>     - Getter methods for **size** and **colour** attributes, and getter and setter method for **currTemp** attribute.

Reviewer: consider splitting this into two lines, one for getters and one for the setter

You need to add the following items:

- Methods:
    - a `toString` method which returns the string `"The z Oven with size of y has the tempeture of x` where x is
      the value of **currTemp**, y is the value of **size**, and z is the value of **colour**.

>     - Two `powerDecrease` and `powerIncrease` methods where the `powerDecrease` decrease the value of **currTemp** by 10

Reviewer: ```suggestion
    - Two methods called `powerDecrease` and `powerIncrease` where the `powerDecrease` method decreases the value of **currTemp** by 10
```

>       and the `powerIncrease` method increase the value of **currTemp** by 10.

Reviewer: ```suggestion
      and the `powerIncrease` method increases the value of **currTemp** by 10.
```

>     - A `changePower` method which call the `powerDecrease` if the value of **currTemp** attribute is more than 100, and

Reviewer: ```suggestion
    - A `changePower` method which call the `powerDecrease` method if the value of the **currTemp** attribute is more than 100, and
```

# oop-special_class_method-medium-q11

## Question Text

> Many appliances and electronic devices have an internal mechanism to moderate temperature. Make a Java class called

Reviewer: get rid of duplicate class information

> **Oven**.

Reviewer 2: ```suggestion
**Oven** to model this using the requirements below:
```

### Oven Class:

Your `Oven` class already has the following items:

- Attributes:
    - A `double` called **currTemp**
    - A `double` called **size**
    - A `String` called **colour**
    - These are not visible from other classes

- Constructor:
    - An `Oven` constructor that is visible to other classes and accepts two `double` and a `String` and uses it to
      initialize the **currTemp**, **size**, and **colour** attributes.
  
- Methods:
    - Getter methods for **size**, **colour**, and **currTemp** attributes.
    - Setter method for **currTemp** attribute.

You need to add the following items:

- Methods:
    - a `toString` method which returns the string `"The z Oven with size of y has the tempeture of x` where x is
      the value of **currTemp**, y is the value of **size**, and z is the value of **colour**.
    - Two methods called `powerDecrease` and `powerIncrease` where the `powerDecrease` method decreases the value of **currTemp** by 10
      and the `powerIncrease` method increases the value of **currTemp** by 10.
    - A `changePower` method which call the `powerDecrease` method if the value of the **currTemp** attribute is more than 100, and

>       call the `powerIncrease` otherwise.

Reviewer 2: The return type of these methods needs to be specified to the students.
